test/functional/test_framework/util.py

```diff
@@ -308,6 +308,15 @@ def get_bip9_status(node, key):
     info = node.getblockchaininfo()
     return info['bip9_softforks'][key]
 
+def get_buried_softforks_status(node, key):
+    for fork in node.getblockchaininfo()['softforks']:
+        if fork['id'] == key:
+            if fork['type'] == "ism":
+                return fork['reject']['status']
+            else:
+                return fork['active']
+    assert(not "unknown softfork")
```

Comment:
I don't understand what this is supposed to be doing. This will always assert if we drop down to this line (a string has truthiness `True`, so `not "string"` is always `False`)

Reply:
It would drop down to this line only if `key` is not found in the results of `getblockchaininfo()`. I want it to exit with the error "unknown softfork" printed

Reply:
ok, in that case you can be much more compact:

```python
def get_buried_softforks_status(node, key):
    softforks = [sf for sf in node.getblockchaininfo()['softforks'] if sf['id'] == key]
    assert softforks, "get_buried_softforks_status() called with invalid softfork %s" % key
    return softforks[0]['active']
```

(assuming you make the change from `['reject']['status']` with `['active']` suggested above)